src/elements/e-table.vue

```diff
@@ -1,67 +1,205 @@
+            :class="b('header-cell', { hidden: true })"
         >
-          <!-- Adding the support for functions was needed, to change visibility state dynamically (improved a11y) -->
-          <span :class="isHeaderLabelVisible(column) ? 'invisible' : null">
-            <!-- Adding the support for functions was needed, to allow the use of translations in the header row -->
-            {{ typeof column.title === 'function' ? column.title() : column.title }}
+          <span class="invisible">
+            {{ $t('e-table.showDetailsHeader') }}
           </span>
-          <button
-            v-if="column.sortable"
-            :class="b('button-sort', getSortButtonModifiers(column))"
-            :aria-label="$t('e-table.sort', { name: column.title })"
-            @click="onClickSort(column)"
-          ></button>
         </th>
       </tr>
     </slot>
-    <tr
-      v-for="(item, itemIndex) in itemsSorted"
-      :key="itemIndex"
-      :class="b('data-row', { disabled: item.disabled })"
-    >
-      <td
-        v-for="(column, columnIndex) in columns"
-        :key="columnIndex"
-        :class="b('data-cell', { text: column.align || 'left', hasEvent: !!column.onClick, col: column.key })"
-        @click="column.onClick ? column.onClick(item, column) : null"
+    <template v-for="(item, itemIndex) in itemsSorted">
+
+      <tr
+        :key="itemIndex"
+        :class="b('data-row', { disabled: item.disabled })"
       >
-        <!-- @slot Use this dynamic slot to add custom templates to the cells -->
-        <slot
-          :name="column.slotName"
-          :item="item"
-          :column="column"
+        <td v-if="selectable"
+            :class="b('data-cell')"
         >
-          {{ item[column.key] }}
-        </slot>
-      </td>
-    </tr>
+          <div :class="b('select-column')">
+            <e-checkbox v-model="selectedInternal"
+                        :value="item[itemIdentifier]"
+                        :name="`e-table__selection--${uuid}`"
+                        :disabled="disabled"
+            >
+              <span :class="{ invisible : !isMobile}">
+                {{ $t('e-table.selectItem') }}
+              </span>
+            </e-checkbox>
+          </div>
+        </td>
+        <td
+          v-for="(column, columnIndex) in columns"
+          :key="columnIndex"
+          :class="b('data-cell', cellModifiers(column))"
+          :data-label="columnTitle(column)"
+          @click="column.onClick || rowHref ? onCellClick(item, column, $event) : null"
+        >
+          <a v-if="!column.onClick && typeof rowHref === 'function'"
+             :class="b('cell-link')"
+             :href="rowHref(item) || '#'"
+             :title="rowTitle(item)"
+             tabindex="-1"
+          ></a>
+
+          <!-- @slot The default 'date' cell formatting for the project. Can be overwritten by a locale <template #date> -->
+          <slot v-if="column.slotName === 'date'"
```

Comment:
Hmm for me this logic is not very clear, especially if we assume multiple columns could be of type `date`. Using a `type` property waa not an option?

Reply:
@MB-web it is not the type. It is the name of the slot. We create dynamic slots based on the `slotName`. You then can give it any cell template you want. This is just an example for a predefined template for general use, when working with `date`. Still, it would be possible to define your own `date` slot-template from outside on usage.